pkg/controller/build/build_controller.go

```diff
@@ -115,10 +113,21 @@ func (r *ReconcileBuild) Reconcile(request reconcile.Request) (reconcile.Result,
 
 	// Everytime control enters the reconcile loop, we need to ensure
 	// everything is in its desired state.
-	buildStrategyInstance := r.retrieveCustomBuildStrategy(instance, request)
-	if buildStrategyInstance != nil {
-		generatedTask = getCustomTask(instance, buildStrategyInstance)
-		generatedTaskRun = getCustomTaskRun(instance, buildStrategyInstance)
+	if instance.Spec.StrategyRef.Kind == buildv1alpha1.NamespacedBuildStrategyKind {
+		buildStrategyInstance := r.retrieveCustomBuildStrategy(instance, request)
+		if buildStrategyInstance != nil {
+			generatedTask = getCustomTask(instance, buildStrategyInstance.Spec.BuildSteps)
+			generatedTaskRun = getCustomTaskRun(instance)
+		}
+	} else if instance.Spec.StrategyRef.Kind == buildv1alpha1.ClusterBuildStrategyKind {
+		clusterBuildStrategyInstance := r.retrieveCustomClusterBuildStrategy(instance, request)
+		if clusterBuildStrategyInstance != nil {
+			generatedTask = getCustomTask(instance, clusterBuildStrategyInstance.Spec.BuildSteps)
```

Comment:
Fair enough, we are creating a namespace'd `Task` ( from a cluster-scoped `BuildStrategy` ) and then an obviously namespace'd `TaskRun` because they are both ownerRef'd to the `Build`.

As a general code improvement, in future, we might want to drop creating the `Task` altogether and create only a `TaskRun` with inline `Task` details? What do you think?

Reply:
Yes, I think there are some duplicated work between Build and Task right now.

My two cents:
1, I don't see too much value for the generated Task right now. 
- If we want to change the build config, we can modify the Build then re-run to re-build
- If we just want to re-build, then we just need to trigger a new BuildRun/TaskRun

2, I think Task is also a template like Build, in our logic, we don't have too much chance to operate or control the Task(middle template)
We can Build then run it by TaskRun directly I think.

For different application, we need a template (Build or Task), but for the one application, I think we don't need two templates :)